### PR TITLE
Hide unreported datasets

### PR DESCRIPTION
**Pull request by fm3**
### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- backend: dataset json api should include new field `isUnreported` (true iff datastores unreported this dataset (“no longer avaible on the datasetore”)
- frontend:
  - log in as admin and use the filtering settings next to the search box in the datasets tab
  - log in as non-admin. the settings box should be hidden. as before, the table should only show datasets which have datalayers.

### Issues:
- first part of #3773 

------
- [x] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)
- ~~[ ] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~~
- ~~[ ] Updated [documentation](../blob/master/docs) if applicable~~
- ~~[ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes~~
- ~~[ ] Needs datastore update after deployment~~
- [x] Ready for review

### CODE REVIEW

**Comment by philippotto** [comment]
@fm3 Can you double check the other checkboxes of the PR's template? For example, do we need to adapt wk-connect, too?

@daniel-wer The front-end is ready for review :)

**Comment by fm3** [comment]
sure, no datastore changes here, the wK backend decides what counts as “unreported” :) I guess there is also no need to update the docs, right?

File: frontend/javascripts/test/snapshots/public/test-bundle/test/backend-snapshot-tests/misc.e2e.js.md [hunk 1]
```diff
@@ -41,17 +41,6 @@ Generated by [AVA](https://ava.li).
       ],
     }
 
-## misc-datastores
```
Comment (daniel-wer) [inline frontend/javascripts/test/snapshots/public/test-bundle/test/backend-snapshot-tests/misc.e2e.js.md]: This doesn't seem right, why is there no datastore?

File: frontend/javascripts/test/snapshots/public/test-bundle/test/backend-snapshot-tests/tasks.e2e.js.md [hunk 2]
```diff
@@ -56,260 +56,9 @@ Generated by [AVA](https://ava.li).
...
-            {
-              id: '59882b370d889b84020efd6f',
-              isTeamManager: true,
-              name: 'team_X4',
-            },
-          ],
-        },
-      },
-    ]
+    []
```
Comment (daniel-wer) [inline frontend/javascripts/test/snapshots/public/test-bundle/test/backend-snapshot-tests/tasks.e2e.js.md]: This also doesn't seem right, or did the PR change something in this regard?

Comment (fm3) [reply frontend/javascripts/test/snapshots/public/test-bundle/test/backend-snapshot-tests/tasks.e2e.js.md]: I re-ran the snapshot tests on my machine and things look better now :thinking:

**Pull request merged**